File: LeetCode/py/1221.py
#Link: https://leetcode.com/problems/split-a-string-in-balanced-strings/

class Solution:
    def balancedStringSplit(self, s: str) -> int:
        left = right = count = 0
        for ch in s:
            if ch == 'L': left += 1
            else: right += 1
            if left == right:
                left = right = 0
                count += 1
        return count
        

File: LeetCode/cpp/1221.cpp
//Link: https://leetcode.com/problems/split-a-string-in-balanced-strings/

class Solution {
public:
    int balancedStringSplit(string s) {
        int left = 0, right = 0, count = 0;
        for(char ch : s) {
            (ch == 'L') ? left += 1 : right += 1;
            if(left == right) {
                left = right = 0;
                count += 1;
            }
        }
        return count;
    }
};
